File: otp_req.py
# # is requested by the php file to call tally and ask for data
import requests
import sys
apiKey = "50900ad3-d2e6-11e8-a895-0200cd936042"
# print(apiKey)
method = sys.argv[1]
if(method=="request_otp"):
    num = sys.argv[2]
    r = requests.post("https://2factor.in/API/V1/{0}/SMS/{1}/AUTOGEN/otp_ledger_form".format(apiKey,num))
    if r.text == "":
        print("there's an error ")
    else:
        print(r.text)
elif(method=="verify_otp"):
    ses = sys.argv[2]
    otp = sys.argv[3]
    # apiKey = "50900ad3-d2e6-11e8-a895-0200cd936042"
    check_url = 'https://2factor.in/API/V1/{0}/SMS/VERIFY/{1}/{2}'.format(apiKey,ses,otp)
    # print(check_url)
    r = requests.post(check_url)
    if r.text == "":
        print("there's an error ")
    else:
        print(r.text)

File: otp_req.php
<?php 
$method = $_GET['method'];
// echo $name;
if($method=="request_otp"){
    $num = $_GET['num'];
    $string = ('py.exe C:\inetpub\wwwroot\otp_req.py '.' '.escapeshellarg($method).' '.escapeshellarg($num));
    $ans = shell_exec($string);
    echo $ans;
}
elseif($method=="verify_otp"){
    $ses = $_GET['ses'];
    $otp = $_GET['otp'];
    // echo $name;
    $string = ('py.exe C:\inetpub\wwwroot\otp_req.py '.escapeshellarg($method).' '.escapeshellarg($ses) .' '.escapeshellarg($otp));
    $ans = shell_exec($string);
    echo $ans;    
}
?>